python/ray/experimental/channel/hccl_group.py

```diff
@@ -0,0 +1,168 @@
+import logging
+from typing import TYPE_CHECKING, List, Optional, Tuple
+import os
+
+import ray
+from ray.exceptions import RayChannelError
+from ray.experimental.channel.gpu_communicator import GPUCommunicator,
+
+
+if TYPE_CHECKING:
+    import torch
+
+
+logger = logging.getLogger(__name__)
+
+
+class _HcclGroup(GPUCommunicator):
+    """
+    Represents an actor's HCCL communicator using NPUs.
+    
+    This is the default HCCL communicator to be used in aDAG if a custom communicator is not provided.
+
+    This class is not thread-safe.
+    """
+
+    def __init__(
+        self,
+        world_size: int,
+        comm_id: int,
+        rank: Optional[int],
+        actor_handles: List["ray.actor.ActorHandle"],
+        device_id: Optional[int],
+    ):
+        """
+        Initialize an HCCL communicator that can be used to communicate p2p with
+        other NPU actors.
+
+        This method blocks until the same call has been made on all other
+        actors in the group, with the same arguments for world_size and comm_id.
+
+        Args:
+            world_size: The number of participating actors/devices.
+            comm_id: A unique communicator ID.
+            rank: The rank of this actor. If None, then the caller is not a
+                participant of the HCCL group.
+            actor_handles: A list of actor handles, in rank order.
+            device_id: The NPU device id to use for HCCL operations.
+        """
+        self._world_size = world_size
+        self._rank: Optional[int] = rank
+        self._actor_handles = actor_handles
+        self._device_id = device_id
+
+        if rank is not None:
+            assert ray.get_gpu_ids(), "HCCL actor has no NPUs assigned"
```

Comment:
`ray.get_gpu_ids()` seems to only get GPU IDs?

Reply:
True, I just changed it. Also I think there should be a API to get all Accelerator id?